allennlp/training/optimizers.py

```diff
@@ -31,143 +31,332 @@
 logger = logging.getLogger(__name__)
 
 
+def make_parameter_groups(
+    model_parameters: List[torch.nn.Parameter],
+    groups: List[Tuple[List[str], Dict[str, Any]]] = None,
+):
```

Comment:
Return type and brief doc string would be nice here.

Reply:
In moving some comments around to make this better, I noticed that I may have messed up some of the implementation.  Our tests should have caught it, though, so either the comment is wrong or I missed something.  Going to double check that (and add a new test if it is in fact broken) before merging this.

Reply:
👍

Reply:
[This test](https://github.com/allenai/allennlp/blob/ff3ed1c89ed10e55da5532ad3fcde9d89747d431/allennlp/tests/training/optimizer_test.py#L39-L62) checks for what I thought was off in the comment.  The test passes, so the comment is wrong.  Merging after adding an updated docstring.